portal-impl/src/com/liferay/portal/defaultpermissions/util/PortalDefaultPermissionsUtil.java

```diff
@@ -53,6 +64,59 @@ public static void setModelDefaultPermissions(
 
 		ResourceLocalServiceUtil.updateModelResources(
 			auditedModel, serviceContext);
+
+		PortalDefaultPermissionsModelResourceRegistry
+			portalDefaultPermissionsModelResourceRegistry =
+				PortalDefaultPermissionsModelResourceRegistryUtil.
+					getPortalDefaultPermissionsModelResourceRegistry();
+
+		PortalDefaultPermissionsModelResource
+			portalDefaultPermissionsModelResource =
+				portalDefaultPermissionsModelResourceRegistry.
+					getPortalDefaultPermissionsModelResource(
+						auditedModel.getModelClassName());
+
+		if (portalDefaultPermissionsModelResource.
+				isAllowOverridePermissions()) {
+
+			_removeResourcePermission(
+				auditedModel, companyId,
+				ResourceActionsUtil.getModelResourceGuestDefaultActions(
+					auditedModel.getModelClassName()),
+				RoleConstants.GUEST, defaultPermissions);
+
+			_removeResourcePermission(
+				auditedModel, companyId,
+				ResourceActionsUtil.getModelResourceGroupDefaultActions(
+					auditedModel.getModelClassName()),
+				RoleConstants.SITE_MEMBER, defaultPermissions);
+
+			_removeResourcePermission(
+				auditedModel, companyId,
+				ResourceActionsUtil.getModelResourceOwnerDefaultActions(
+					auditedModel.getModelClassName()),
+				RoleConstants.OWNER, defaultPermissions);
+		}
+	}
+
+	private static void _removeResourcePermission(
+			AuditedModel auditedModel, long companyId,
+			List<String> defaultActions, String roleName,
+			Map<String, String[]> defaultPermissions)
+		throws PortalException {
+
+		List<String> actionIds = Arrays.asList(
+			defaultPermissions.getOrDefault(roleName, new String[0]));
+
+		Role role = RoleLocalServiceUtil.fetchRole(companyId, roleName);
```

Comment:
probably we can use the get (also because if the role remains null we will have a NPE on line 116)